packages/protocol/contracts/common/UniswapFeeHandlerSeller.sol

```diff
@@ -0,0 +1,194 @@
+  uint256 constant MAX_TIMESTAMP_BLOCK_EXCHANGE = 20;
+
+  event ReceivedQuote(address router, uint256 quote);
+  event RouterUsed(address router);
+  event RouterAddressSet(address token, address router);
+  event RouterAddressRemoved(address token, address router);
+
+  /**
+   * @notice Sets initialized == true on implementation contracts.
+   * @param test Set to true to skip implementation initialisation.
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  // without this line the contract can't receive native Celo transfers
+  function() external payable {}
+
+  /**
+   * @notice Returns the storage, major, minor, and patch version of the contract.
+   * @return Storage version of the contract.
+   * @return Major version of the contract.
+   * @return Minor version of the contract.
+   * @return Patch version of the contract.
+   */
+  function getVersionNumber() external pure returns (uint256, uint256, uint256, uint256) {
+    return (1, 1, 0, 0);
+  }
+
+  /**
+    * @notice Allows owner to set the router for a token.
+    * @param token Address of the token to set.
+    * @param router The new router.
+    */
+  function setRouter(address token, address router) external onlyOwner {
+    _setRouter(token, router);
+  }
+
+  function _setRouter(address token, address router) private {
+    require(router != address(0), "Router can't be address zero");
+    routerAddresses[token].push(router);
+    emit RouterAddressSet(token, router);
+  }
+
+  /**
+    * @notice Allows owner to remove a router for a token.
+    * @param token Address of the token.
+    * @param router Address of the router to remove.
+    * @param index The index of the router to remove.
+    */
+  function removeRouter(address token, address router, uint256 index) external onlyOwner {
+    require(routerAddresses[token][index] == router, "Index does not match");
+
+    uint256 length = routerAddresses[token].length;
+    routerAddresses[token][index] = routerAddresses[token][length - 1];
+    routerAddresses[token].pop();
+    emit RouterAddressRemoved(token, router);
+  }
+
+  /**
+    * @notice Get the list of routers for a token.
+    * @param token The address of the token to query.
+    * @return An array of all the allowed router.
+    */
+  function getRoutersForToken(address token) external view returns (address[] memory) {
+    return routerAddresses[token];
+  }
+
+  function calculateAllMinAmount(
+    address sellTokenAddress,
+    uint256 maxSlippage,
+    uint256 amount,
+    IUniswapV2RouterMin bestRouter
+  ) private returns (uint256) {
+    ISortedOracles sortedOracles = getSortedOracles();
+    require(
+      sortedOracles.numRates(sellTokenAddress) >= minimumReports[sellTokenAddress],
+      "Number of reports for token not enough"
+    );
+
+    uint256 minimalSortedOracles = 0;
+    // if minimumReports for this token is zero, assume the check is not needed
+    if (minimumReports[sellTokenAddress] > 0) {
```

Comment:
Small gas improvement:

Since we are using `minimumReports[sellTokenAddress]` 2 times we can save it to local variable